### PR TITLE
refactor: Add `@dxos/echo-protocol` and `@dxos/kv-store`

### PR DESCRIPTION
**Pull request by mykola-vrmchk**
### Motivation / Background

This Pull Request ...

### CODE REVIEW

**Comment by mykola-vrmchk** [comment]
* **#6527** <a href="https://app.graphite.dev/github/pr/dxos/dxos/6527?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @mykola-vrmchk and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/core/echo/echo-protocol/eslint-config.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// TODO(wittjosiah): This is not currently being used because testing packlet breaks the packlet lint.
```
Comment (dmaretskyi) [inline packages/core/echo/echo-protocol/eslint-config.json:1]: is this only for packlets? remove if so

File: packages/core/echo/echo-protocol/package.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  "version": "0.5.0",
+  "description": "Core ECHO APIs.",
+  "homepage": "https://dxos.org",
+  "bugs": "https://github.com/dxos/dxos/issues",
+  "repository": "github:dxos/dxos",
+  "license": "MIT",
+  "author": "DXOS.org",
+  "main": "dist/lib/node/index.cjs",
+  "browser": {
+    "./dist/lib/node/index.cjs": "./dist/lib/browser/index.mjs"
```
Comment (dmaretskyi) [inline packages/core/echo/echo-protocol/package.json:12]: use "exports" like in the kv-store package

File: packages/core/echo/echo-protocol/project.json [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    "scope:echo"
+  ],
+  "sourceRoot": "packages/core/echo/echo-protocol/src",
+  "projectType": "library",
+  "targets": {
+    "build": {},
+    "compile": {
+      "options": {
+        "entryPoints": [
+          "packages/core/echo/echo-protocol/src/index.ts"
```
Comment (dmaretskyi) [inline packages/core/echo/echo-protocol/project.json:14]: nx recently added support for `{projectRoot}` in configs - make use of it to make configs uniform

File: packages/core/echo/echo-protocol/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
+{
+  "extends": "../../../../tsconfig.json",
+  "compilerOptions": {
+    "outDir": "dist/types",
+    "types": [
+      "@dxos/test",
+      "@dxos/typings"
```
Comment (dmaretskyi) [inline packages/core/echo/echo-protocol/tsconfig.json:7]: are those used?

File: packages/common/kv-store/tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,21 @@
+{
+  "extends": "../../../tsconfig.json",
+  "compilerOptions": {
+    "lib": [
+      "DOM"
```
Comment (dmaretskyi) [inline packages/common/kv-store/tsconfig.json:5]: do we need dom?

File: packages/common/kv-store/project.json [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    "scope:common"
+  ],
+  "sourceRoot": "packages/common/kv-store/src",
+  "projectType": "library",
+  "targets": {
+    "build": {},
+    "compile": {
+      "options": {
+        "entryPoints": [
+          "packages/common/kv-store/src/index.ts",
```
Comment (dmaretskyi) [inline packages/common/kv-store/project.json:14]: {projectRoot} -- see other comment

File: packages/core/echo/echo-protocol/eslint-config.json (hunk 1 above)
Comment (mykola-vrmchk) [reply packages/core/echo/echo-protocol/eslint-config.json:1]: We need to do it in other packages also

**Pull request merged**